tools/clang/lib/Sema/SemaChecking.cpp

```diff
@@ -7085,6 +7086,13 @@ void CheckImplicitConversion(Sema &S, Expr *E, QualType T,
   if (auto VecTy = dyn_cast<VectorType>(Target))
     Target = VecTy->getElementType().getTypePtr();
 
+  // HLSL Change Begins: Vector/matrix implicit conversions already diagnosed
+  if (hlsl::IsVectorType(&S, QualType(Source, 0)) || hlsl::IsMatrixType(&S, QualType(Source, 0)) ||
```

Comment:
It hits the codepath for vector type here, though it doesn't show a difference.  It cuts the code off from trying to handle cases that are not implemented here. Though, I just thought of a slight optimization I could do here: check for struct type first, so it would trivially skip common types that could not be HLSL matrix/vector types.

Reply:
Ok.